src/Microsoft.Crank.Controller/Program.cs

```diff
@@ -3351,7 +3388,11 @@ private static async Task<string> GetRelayTokenAsync(Uri endpointUri)
             }
 
             var rcsb = new RelayConnectionStringBuilder(connectionString);
-            var tokenProvider = TokenProvider.CreateSharedAccessSignatureTokenProvider(rcsb.SharedAccessKeyName, rcsb.SharedAccessKey);
+            TokenProvider tokenProvider = null;
+            AccessToken token = default;
+            
+            tokenProvider = TokenProvider.CreateSharedAccessSignatureTokenProvider(rcsb.SharedAccessKeyName, rcsb.SharedAccessKey);
```

Comment:
This should make use of GetAadTokenBuilder, and I think we should do the same for the service bus, perhaps we can make the TokenProvider code reusable across the projects?

Reply:
What is GetAadTokenBuilder, did you mean GetAadTokenProvider?

Reply:
ah yes, I meant `GetAadTokenProvider `

Reply:
So as I mentioned in one of Sebastien's comments, I removed the use of certs for the Controller in Azure Relay. This is because, due to conditional access policy, we can only use the service principal authentication story on machines that can access the machines on the relay. So for now I was thinking that we would not change this behavior.